drivers/flash/flash_mcux_flexspi_nor.c

```diff
@@ -49,53 +52,98 @@ struct flash_flexspi_nor_data {
 	},
 
-	[ERASE_SECTOR] = {
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR,       kFLEXSPI_1PAD, SPI_NOR_CMD_SE,
-				kFLEXSPI_Command_RADDR_SDR, kFLEXSPI_1PAD, 0x18),
+	/* Write Enable */
+	[NOR_CMD_LUT_SEQ_IDX_WRITEENABLE] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_WREN,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
 	},
 
-	[ERASE_CHIP] = {
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR,       kFLEXSPI_1PAD, SPI_NOR_CMD_CE,
-				kFLEXSPI_Command_STOP,      kFLEXSPI_1PAD, 0),
+	/* Erase Sector  */
+	[NOR_CMD_LUT_SEQ_IDX_ERASESECTOR] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_SE,
+				kFLEXSPI_Command_RADDR_SDR, kFLEXSPI_1PAD, 0x18),
 	},
 
-	[READ_FAST_QUAD_OUTPUT] = {
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR,       kFLEXSPI_1PAD, 0x6B,
+	/* Page Program - single mode */
+	[NOR_CMD_LUT_SEQ_IDX_PAGEPROGRAM_SINGLE] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_PP,
 				kFLEXSPI_Command_RADDR_SDR, kFLEXSPI_1PAD, 0x18),
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DUMMY_SDR, kFLEXSPI_4PAD, 0x08,
-				kFLEXSPI_Command_READ_SDR,  kFLEXSPI_4PAD, 0x04),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_WRITE_SDR, kFLEXSPI_1PAD, 0x04,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
 	},
 
-	[PAGE_PROGRAM_QUAD_INPUT] = {
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR,       kFLEXSPI_1PAD, 0x32,
+	/* Page Program - quad mode */
+	[NOR_CMD_LUT_SEQ_IDX_PAGEPROGRAM_QUAD] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, 0x32,
 				kFLEXSPI_Command_RADDR_SDR, kFLEXSPI_1PAD, 0x18),
 		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_WRITE_SDR, kFLEXSPI_4PAD, 0x04,
-				kFLEXSPI_Command_STOP,      kFLEXSPI_1PAD, 0),
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
+	},
+
+	/* Read ID */
+	[NOR_CMD_LUT_SEQ_IDX_READID] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_RDID,
+				kFLEXSPI_Command_READ_SDR, kFLEXSPI_1PAD, 0x04),
+	},
+
+	/* Enable Quad mode */
+	[NOR_CMD_LUT_SEQ_IDX_WRITESTATUSREG] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_WRSR,
+				kFLEXSPI_Command_WRITE_SDR, kFLEXSPI_1PAD, 0x04),
+	},
+
+	/* Enter QPI mode */
+	[NOR_CMD_LUT_SEQ_IDX_ENTERQPI] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, 0x35,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
 	},
 
-	[ENTER_QPI] = {
-		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR,       kFLEXSPI_1PAD, 0x35,
-				kFLEXSPI_Command_STOP,      kFLEXSPI_1PAD, 0),
+	/* Exit QPI mode */
+	[NOR_CMD_LUT_SEQ_IDX_EXITQPI] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_4PAD, 0xF5,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
+	},
+
+	/* Read status register */
+	[NOR_CMD_LUT_SEQ_IDX_READSTATUSREG] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_RDSR,
+				kFLEXSPI_Command_READ_SDR, kFLEXSPI_1PAD, 0x04),
+	},
+
+	/* Erase whole chip */
+	[NOR_CMD_LUT_SEQ_IDX_ERASECHIP] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, SPI_NOR_CMD_CE,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
```

Comment:
It's hard to tell which instructions actually changed after renaming and reordering them, and changing the whitespace too. Can you restore the original names and order so we can see more clearly where instructions changed?

Reply:
Sure, I'll try to get that cleaner. However can the enumeration of the instructions be changed? I followed this [NXP article](https://community.nxp.com/t5/i-MX-RT-Knowledge-Base/How-to-use-the-RT1064-on-chip-flash-as-NVM/ta-p/1123381) where it was explicitly mentioned to align the order to XIP. Or do you mean the order in which they are set in the lookup table itself?